src/python/espressomd/visualization_opengl.pyx

```diff
@@ -783,19 +789,47 @@ class openGLLive:
         # Collect shapes and interaction type (for coloring) from constraints
         primitive_shapes = [
             'Shapes::Wall', 'Shapes::Cylinder', 'Shapes::Ellipsoid',
-            'Shapes::SimplePore', 'Shapes::Slitpore', 'Shapes::Sphere',
-            'Shapes::SpheroCylinder']
+            'Shapes::HollowConicalFrustum', 'Shapes::SimplePore',
+            'Shapes::Slitpore', 'Shapes::Sphere', 'Shapes::SpheroCylinder']
+
+        def get_shapes_from_union(union):
```

Comment:
if you want to hide the special case of the union, you could differentiate between the union and the others within this function. You could also probably get rid of ```update_shapes_collection``` and fail late. Just try drawing the ```HollowConicalFrustum``` with GLE and if you fail fall back to ```Shapes::Misc```.